### PR TITLE
Feature/1.x webhook support

### PR DESCRIPTION
**Pull request by jerbob92**
I created a basic webhook notification implementation so people can make Flapjack integration without messing with the Flapjack source.

The webhook sends the alert details in json to the URL. Multiple URLs can be configured. 
Other actions can then be done with the Flapjack API.

I did not create this in the 2.x branch as contacts seem pretty broken in there. Maybe it can easily be merged into 2.x though.

Here is a basic example to receive the notifications in NodeJS (requires express) which works with the example configuration:

```javascript
var express = require('express'),
    app = express();
app.use(express.bodyParser());
app.use(express.logger("default"));

var callback = function (req, res) {
    console.log(req.body);
};

app.post('/flapjack_notification', callback);

app.listen(80);
```

### CODE REVIEW

File: etc/flapjack_config.yaml.example [hunk 1]
```diff
@@ -247,6 +248,15 @@ production:
       logger:
         level: INFO
         syslog_errors: yes
+    webhook:
+      enabled: no
+      queue: webhook_notifications
+      hooks:
+        - url: "http://127.0.0.1/flapjack_notification"
```
Comment (ali-graham) [inline etc/flapjack_config.yaml.example]: This is an externally provided webhook destination for testing?

**Comment by ali-graham** [comment]
Looks good for v1.x, and I can handle the port to master. This may all be revisited for v3.0, where we hope to change the gateway interface to make it easier for people to publish them as third-party gems (something that was on the v2.0 roadmap but has been deferred).

@jessereynolds What do you think?

File: etc/flapjack_config.yaml.example (hunk 1 above)
Comment (ali-graham) [inline etc/flapjack_config.yaml.example]: Ah, yup, I see the code above; sorry, moving slowly this morning. :)

**Comment by jessereynolds** [comment]
This is great! Thanks @jerbob92

**Comment by jerbob92** [comment]
That was quick, thanks!
Pluginnable notifiers through gems would be very nice!
Auto detection of plugins would also be great, I had a hard time figuring out which parts to edit to make the notifier work.

Something like this to get the notifiers (where all notifiers are classes in the module Flapjack::Notifiers:
```ruby
Flapjack::Notifiers.constants.select {|c| Class === Flapjack::Notifiers.const_get(c)}
```

Then in the plugin we can have something like:
```ruby
module Flapjack
  module Notifiers
    class Webhook
      attr_accessor :configuration_key, :media_options
      def initialize(opts = {})
        @configuration_key = "webhook"
        @media_options = {
          "username" => {
            "type" => "textfield",
            "label" => "Username"
           }
         }
      end
    end
  end
end
```

The media_options would allow for a more dynamic media configuration form.

**Pull request merged**